src/lib.rs

```diff
@@ -25,6 +25,40 @@ use std::ops::{Deref, DerefMut};
 /// Determines how a memory map may be used. If the memory map is backed by a
 /// file, then the file must have permissions corresponding to the operations
 /// the protection level allows.
+///
+/// # Example
+///
+/// ```rust
+/// # use std::error::Error;
+/// use std::fs::File;
+/// use std::io::Write;
+/// use memmap::Protection;
+///
+/// # fn try_main() -> Result<(), Box<Error>> {
+/// let file = File::open("README.md")?;
+///
+/// // Initialize a mutable memory map with `ReadCopy` protection
+/// let mut mmap = unsafe { memmap::file(&file)
+///                             .protection(Protection::ReadCopy)
+///                             .map_mut()? };
+///
+/// // This is allowed
+/// (&mut mmap[..]).write(b"Hello")?;
+///
+/// // This is also allowed
+/// mmap.set_protection(Protection::ReadWrite)?;
+///
+/// // This is NOT allowed and will return an error,
+/// // since we are trying to make a `MmapMut` read-only.
+/// mmap.set_protection(Protection::Read)?;
```

Comment:
See #46 for what the API might look like.  I'm going to merge yours first since that PR will take a while to land, so don't worry about potentia conflicts.